pytorch_lightning/trainer/trainer.py

```diff
@@ -2016,7 +2016,11 @@ def should_rank_save_checkpoint(self) -> bool:
 
     @property
     def num_nodes(self) -> int:
-        return self._accelerator_connector.num_nodes
+        rank_zero_deprecation(
+            "`Trainer.num_nodes` was deprecated in v1.6 and will be removed in v1.8."
+            ' Please use `getattr(Trainer.strategy, "num_nodes", 1)` instead.'
+        )
```

Comment:
do we need to deprecate it off the trainer? given it's a constructor argument to the Trainer, this is very convenient for users.

fully agreed with removing the duplicated property from the accelerator connector though

Reply:
Discussed on the issue, will not deprecate this off the trainer.